src/NuGet.Core/NuGet.PackageManagement/BuildIntegration/DependencyGraphRestoreUtility.cs

```diff
@@ -132,34 +132,14 @@ public static class DependencyGraphRestoreUtility
 
                     RestoreSummary.Log(log, restoreSummaries);
 
-                    await PersistDGSpec(dgSpec);
-
                     return restoreSummaries;
                 }
             }
 
             return new List<RestoreSummary>();
         }
 
-        private static async Task PersistDGSpec(DependencyGraphSpec dgSpec)
-        {
-            try
-            {
-                var filePath = GetDefaultDGSpecFileName();
-
-                // create nuget temp folder if not exists
-                DirectoryUtility.CreateSharedDirectory(Path.GetDirectoryName(filePath));
-
-                // delete existing dg spec file (if exists) then replace it with new file.
-                await FileUtility.ReplaceWithLock(
-                    (tempFile) => dgSpec.Save(tempFile), filePath);
-            }
-            catch (Exception)
-            {
-                //ignore any failure if it fails to write or replace dg spec file.
-            }
-        }
-
+        [Obsolete]
```

Comment:
Should we link to this PR and/or explain which version of each product is the last one that supported this method?

Reply:
It's an interesting idea to document when obsolete APIs became obsolete, but I feel like this is a process change that is out of scope for this PR.  If you'd like to propose a team process change, and that gets accepted before this PR gets merged, I'll follow the team guideline.

Also note that our [nuget sdk docs](https://github.com/NuGet/NuGet.Client/blob/dev/docs/nuget-sdk.md#api-compatibility-policy) call out NuGet.PackageManagement as a special case. It's closely tied to VS, which frankly should never have been made public in my opinion, but the people who worked on NuGet before us didn't appear to do any public API design, or consider the tech debt of supporting public APIs for people years later 😢 

This API is a good example, I really don't see any use for anyone to ever call it, except for our VS feedback provider. It somehow should have been "internal" to NuGet, not public. Personally, I'd prefer to just delete the API, but it's "protected" by the public API analyzer, and costs little for us to obsolete then ignore forever (or until we come up with a plan to delete public APIs and deal with the consequences of customers it affects).